cryptpad: 2025.3.0 -> 2025.6.0

themadbit opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done
- Updated the package
- Added an updateScript

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

martinetd commented on pkgs/by-name/cr/cryptpad/package.nix:
```diff
@@ -87,7 +129,7 @@ buildNpmPackage {
   };
 
-  npmDepsHash = "sha256-GWkyRlizPSA72WwoY+mRLwaMeD/SXdo6oUVwsd2gp7c=";
+  npmDepsHash = "sha256-4Zr+8ANZJ9XX2umY/SY7BrEHPheVelFSeZipgOaW6bI==";
```
This time the npmDepsHash got updated but versions above was left at 2025.3.0 :grin: 
(from your original patch)
```
diff --git a/pkgs/by-name/cr/cryptpad/package.nix b/pkgs/by-name/cr/cryptpad/package.nix
index 7372e17a8344..2338f5c841ee 100644
--- a/pkgs/by-name/cr/cryptpad/package.nix
+++ b/pkgs/by-name/cr/cryptpad/package.nix
@@ -11,10 +11,11 @@
   nodejs,
   rdfind,
   unzip,
+  nix-update-script,
 }:
 
 let
-  version = "2025.3.0";
+  version = "2025.6.0";
   # nix version of install-onlyoffice.sh
   # a later version could rebuild from sdkjs/web-apps as per
   # https://github.com/cryptpad/onlyoffice-builds/blob/main/build.sh
@@ -126,7 +127,7 @@ buildNpmPackage {
     owner = "cryptpad";
     repo = "cryptpad";
     rev = version;
-    hash = "sha256-NxkVMsfLzdzifdn+f0C6mBJGd1oLwcMTAIXv+gBG7rI=";
+    hash = "sha256-R8Oonrnb1tqvl1zTkWv5Xv/f8bFtUljD6X/re72IvsU=";
   };
 
   npmDepsHash = "sha256-4Zr+8ANZJ9XX2umY/SY7BrEHPheVelFSeZipgOaW6bI==";
```

martinetd commented on pkgs/by-name/cr/cryptpad/package.nix:
```diff
@@ -18,18 +18,55 @@ let
+          mkdir -p "$oo_dir"
+          cd "$oo_dir"
+          unzip ${onlyoffice_newfetch oo.version oo.hash}
+          echo "${oo.version}" > "$oo_dir/.version"
```
This is after `cd $oo_dir` so looks a bit odd.. but the file is present:
```
$ cat result/lib/node_modules/cryptpad/www/common/onlyoffice/dist/v7/.version 
v7.3.3.60+11
```
so I must be missing something

martinetd commented on pkgs/by-name/cr/cryptpad/package.nix:
```diff
@@ -18,18 +18,55 @@ let
+          rm -rf web-apps/apps/documenteditor/main/resources/help
+          rm -rf web-apps/apps/presentationeditor/main/resources/help
+          rm -rf web-apps/apps/spreadsheeteditor/main/resources/help
+          ${lib.optionalString (oo.subdir == "v7") ''
```
we don't want to remove any doc for v8 (install-onlyoffice.sh only does this for old versions)

martinetd commented on pkgs/by-name/cr/cryptpad/package.nix:
```diff
@@ -18,18 +18,55 @@ let
+          echo "${oo.version}" > "$oo_dir/.version"
+
+          # Clean up help files and dictionaries as per upstream (using relative paths since we're in $oo_dir)
+          rm -rf web-apps/apps/documenteditor/main/resources/help
```
(nitpick) this is fine as it is but I think it'd be ok to remove `web-apps/apps/*/main/resources/help` and not bother listing each path individually?

martinetd commented on pkgs/by-name/cr/cryptpad/package.nix:
```diff
@@ -120,8 +162,9 @@ buildNpmPackage {
     mv www/components "$out_cryptpad/www/"
-    # and fix absolute symlink to /build...
-    ln -Tfs ../../src/tweetnacl "$out_cryptpad/www/components/tweetnacl"
+    # Remove the directory first if it exists, then create the symlink
```
I don't think we need to bother with this anymore in this version?
In 2025.3.0 www/components/tweetnacl was a symlink to /build so we had to fix it, but in the new version the directory is properly installed

OTOH the content is identical so symlink is probably fine (so long as cryptpad doesn't do any weird check based on realpath), just that at this point I've checked and all directories in www/components also exist and are identical to node modules so if we do that we might just as well replace them all and not single tweetnacl out
(nix store optimize will fix all the individual files away, but it's 225MB so I assume it'd be better optimized away if it doesn't hurt)

martinetd commented on pkgs/by-name/cr/cryptpad/package.nix:
```diff
@@ -18,18 +18,55 @@ let
+          # New method for v7+ versions
+          mkdir -p "$oo_dir"
+          cd "$oo_dir"
+          unzip ${onlyoffice_newfetch oo.version oo.hash}
```
So I don't see that in the cryptpad install script, but with this we end up with this path:
```
./www/common/onlyoffice/dist/v8/sdkjs/common/serviceworker/document_editor_service_worker.js
```
and after fixing the tests I see a 404 error for this path:
```
vm-test-run-cryptpad> machine # [   17.796145] cryptpad[653]: ["INFO","2025-08-24T02:38:13.244Z","HTTP_404","/common/onlyoffice/dist/v8/document_editor_service_worker.js"]
```
(so same path without `sdkjs/common/serviceworker`)

I think this will need manual testing (or perhaps adding a case to selenium?) and fixing that path too :/

martinetd reviewed (commented):
Thanks for the update!

I've had a look at the test failure, and I'm not sure where it's supposed to be installed but test fails with 404 error on e.g. `common/common-util.js` and other files in cryptpad `src` dir

After adding symlinks from ../../src/common/* to www/common (including files in subdirectories) then at least the selenium test passes, I didn't check further (yet)
```
lrwxrwxrwx 1 nix nix     23 Aug 24 11:35 result/lib/node_modules/cryptpad/www/common/rpc.js -> ../../src/common/rpc.js
lrwxrwxrwx 1 nix nix     26 Aug 24 11:35 result/lib/node_modules/cryptpad/www/common/notify.js -> ../../src/common/notify.js
lrwxrwxrwx 1 nix nix     26 Aug 24 11:35 result/lib/node_modules/cryptpad/www/common/pinpad.js -> ../../src/common/pinpad.js
lrwxrwxrwx 1 nix nix     28 Aug 24 11:35 result/lib/node_modules/cryptpad/www/common/cryptget.js -> ../../src/common/cryptget.js
lrwxrwxrwx 1 nix nix     30 Aug 24 11:35 result/lib/node_modules/cryptpad/www/common/recurrence.js -> ../../src/common/recurrence.js
lrwxrwxrwx 1 nix nix     31 Aug 24 11:28 result/lib/node_modules/cryptpad/www/common/common-hash.js -> ../../src/common/common-hash.js
lrwxrwxrwx 1 nix nix     31 Aug 24 11:28 result/lib/node_modules/cryptpad/www/common/common-util.js -> ../../src/common/common-util.js
lrwxrwxrwx 1 nix nix     31 Aug 24 11:31 result/lib/node_modules/cryptpad/www/common/cache-store.js -> ../../src/common/cache-store.js
lrwxrwxrwx 1 nix nix     31 Aug 24 11:35 result/lib/node_modules/cryptpad/www/common/user-object.js -> ../../src/common/user-object.js
lrwxrwxrwx 1 nix nix     33 Aug 24 11:35 result/lib/node_modules/cryptpad/www/common/proxy-manager.js -> ../../src/common/proxy-manager.js
lrwxrwxrwx 1 nix nix     34 Aug 24 11:35 result/lib/node_modules/cryptpad/www/common/events-channel.js -> ../../src/common/events-channel.js
lrwxrwxrwx 1 nix nix     34 Aug 24 11:35 result/lib/node_modules/cryptpad/www/common/network-config.js -> ../../src/common/network-config.js
lrwxrwxrwx 1 nix nix     35 Aug 24 11:28 result/lib/node_modules/cryptpad/www/common/common-feedback.js -> ../../src/common/common-feedback.js
lrwxrwxrwx 1 nix nix     35 Aug 24 11:28 result/lib/node_modules/cryptpad/www/common/common-realtime.js -> ../../src/common/common-realtime.js
lrwxrwxrwx 1 nix nix     36 Aug 24 11:28 result/lib/node_modules/cryptpad/www/common/common-constants.js -> ../../src/common/common-constants.js
lrwxrwxrwx 1 nix nix     37 Aug 24 11:28 result/lib/node_modules/cryptpad/www/common/common-credential.js -> ../../src/common/common-credential.js
lrwxrwxrwx 1 nix nix     38 Aug 24 11:35 result/lib/node_modules/cryptpad/www/common/user-object-setter.js -> ../../src/common/user-object-setter.js
lrwxrwxrwx 1 nix nix     39 Aug 24 11:28 result/lib/node_modules/cryptpad/www/common/common-signing-keys.js -> ../../src/common/common-signing-keys.js
lrwxrwxrwx 1 nix nix     40 Aug 24 11:35 result/lib/node_modules/cryptpad/www/common/outer/login-block.js -> ../../../src/common/outer/login-block.js
lrwxrwxrwx 1 nix nix     41 Aug 24 11:35 result/lib/node_modules/cryptpad/www/common/outer/http-command.js -> ../../../src/common/outer/http-command.js
lrwxrwxrwx 1 nix nix     49 Aug 24 11:29 result/lib/node_modules/cryptpad/www/common/onlyoffice/current-version.js -> ../../../src/common/onlyoffice/current-version.js
```

themadbit commented on pkgs/by-name/cr/cryptpad/package.nix:
```diff
@@ -144,6 +180,7 @@ buildNpmPackage {
     # directory.
     makeWrapper "${lib.getExe nodejs}" "$out/bin/cryptpad" \
       --add-flags "$out_cryptpad/server.js" \
+      --run "mkdir -p \$HOME/.local/share/cryptpad && cd \$HOME/.local/share/cryptpad" \
```
I will remove this. It's for debugging. --Allows execution of bin in the nixpkgs dir

themadbit commented:
Hi @martinetd 

I have addressed most of the comments. What do you reckon should be the next steps and maybe implementation notes to get this merged? :)

martinetd reviewed (commented):
> I have addressed most of the comments. What do you reckon should be the next steps and maybe implementation notes to get this merged? :)

Thanks for the fixes!

Unfortunately this doesn't work right now (as illustrated by the selenium test not loading the page properly) because of https://github.com/NixOS/nixpkgs/pull/434721/files#r2296449464 - so that would be the next point to address.

After that fixing the install-onlyoffice.sh check for later upgrades would be a bonus, but I reckon you've done enough so that can probably wait for later as far as I'm concerned.

If you don't have time to deal with the broken install then I can take a look and finish this over the weekend probably, maybe next one if not this one.

martinetd commented on pkgs/by-name/cr/cryptpad/package.nix:
```diff
@@ -14,22 +14,53 @@
+          echo "${oo.rev}" > "$oo_dir/.commit"
+
+          # Clean up help files as per upstream
+          rm -rf "$oo_dir/web-apps/apps/*/main/resources/help"
```
Ah, I hadn't noticed but this wildcard is not expanded, the `*` needs to be outside of quotes to work (for example `lib/node_modules/cryptpad/www/common/onlyoffice/dist/v5/web-apps/apps/spreadsheeteditor/main/resources/help/` was not removed)

I've also fixed that in my branch, that saves quite a bit of space (2.9GB -> 2.2GB)

themadbit committed:
cryptpad: 2025.3.0 -> 2025.6.0

Signed-off-by: themadbit <marktanui75@gmail.com>

themadbit commented:
@martinetd thanks. I have picked up and applied the patch. 
From my testing, the interactive test `nix build .#cryptpad.passthru.tests.cryptpad.driverInteractive` builds and runs all tests successfully. However, the non-interactive `nix build .#cryptpad.passthru.tests.cryptpad`, breaks due to some problems with the `webdriver`, and I think the problem might be how the headless Firefox is configured. I am looking into that and will update you.

martinetd commented:
Thanks!

I think there shouldn't be any difference between interactive and non-interactive as far as the test can see, but there might be a race somewhere :/
At least the non-interactive test passes on my machine with your commit...

I'll try to run this for real, but as said above this is probably good enough for now.
Just let me know what you figure out about the test failure first..

martinetd reviewed (approved):
Thank you!

martinetd commented:
(Ugh.. I just noticed the linux-aarch64 test failed :/ . . . but looking at a previous PR it doesn't look like a new failure, some worker fails to spawn and the process exists..
I guess I'll open an issue upstream when time allows and we can ignore that too for now)

martinetd commented:
> Hello, fyi, a new version has been published : [cryptpad/cryptpad@2025.9.0 (release)](https://github.com/cryptpad/cryptpad/releases/tag/2025.9.0)

I'd rather just see this merged first, but ¯\_(ツ)_/¯ 

At least there wasn't any packaging change worth noting there, so this ought to be enough if we want to upgrade (nixosTests.cryptpad passes on my machine):
```
diff --git a/pkgs/by-name/cr/cryptpad/package.nix b/pkgs/by-name/cr/cryptpad/package.nix
index 7fee527aabbc..6088fbcaa6aa 100644
--- a/pkgs/by-name/cr/cryptpad/package.nix
+++ b/pkgs/by-name/cr/cryptpad/package.nix
@@ -14,7 +14,7 @@
 }:
 
 let
-  version = "2025.6.0";
+  version = "2025.9.0";
   # nix version of install-onlyoffice.sh
   # a later version could rebuild from sdkjs/web-apps as per
   # https://github.com/cryptpad/onlyoffice-builds/blob/main/build.sh
@@ -93,8 +93,8 @@ let
     }
     {
       subdir = "v8";
-      version = "v8.3.3.23+4";
-      hash = "sha256-DeK84fa7Jc1L1+vF8LBKLXM5oWS0SV2qBnAWG3Xzu4U=";
+      version = "v8.3.3.23+5";
+      hash = "sha256-+53jzvmGltD1yjXAimLl8zL1V4YDc1qF1PUFSeyiUm8=";
     }
   ];
 
@@ -118,10 +118,10 @@ buildNpmPackage {
     owner = "cryptpad";
     repo = "cryptpad";
     rev = version;
-    hash = "sha256-R8Oonrnb1tqvl1zTkWv5Xv/f8bFtUljD6X/re72IvsU=";
+    hash = "sha256-veLtKjrk1CZe2u3MkozsPK98hyhdsWbQGUxh8oWjLXg=";
   };
 
-  npmDepsHash = "sha256-4Zr+8ANZJ9XX2umY/SY7BrEHPheVelFSeZipgOaW6bI==";
+  npmDepsHash = "sha256-d/2JKGdC/tgDOo4Qr/0g83lh5gW6Varr0vkZUZe+WTA=";
 
   nativeBuildInputs = [
     makeBinaryWrapper
```

eljamm commented on pkgs/by-name/cr/cryptpad/package.nix:
```diff
@@ -14,22 +14,51 @@
+    fetchurl {
+      url = "https://github.com/cryptpad/onlyoffice-editor/releases/download/${version}/onlyoffice-editor.zip";
+      inherit hash;
+    };
```
This can be refactored into a single function. Then, to use it, you can just pass `onlyoffice_fetch oo`.

```suggestion
  onlyoffice_fetch =
    {
      rev ? null,
      version ? null,
      hash,
      ...
    }:
    assert (
      lib.assertMsg (lib.xor (rev == null) (
        version == null
      )) "onlyoffice_fetch requires one of either `rev` or `version` to be provided (not both)."
    );
    if rev != null then
      fetchFromGitHub {
        inherit rev hash;
        owner = "cryptpad";
        repo = "onlyoffice-builds";
      }
    # New method for v7+ versions that use ZIP releases
    else
      fetchurl {
        url = "https://github.com/cryptpad/onlyoffice-editor/releases/download/${version}/onlyoffice-editor.zip";
        inherit hash;
      };
```

eljamm commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 434721`
Commit: [`06b08e536f1d219564f9512563334b913bfbbdb9`](https://github.com/NixOS/nixpkgs/commit/06b08e536f1d219564f9512563334b913bfbbdb9) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/06b08e536f1d219564f9512563334b913bfbbdb9..pull/434721/head))
Merge: [`29716a5c1dc93aca3dc9db0566b9d75afb0be472`](https://github.com/NixOS/nixpkgs/commit/29716a5c1dc93aca3dc9db0566b9d75afb0be472)

Logs: https://github.com/eljamm/nixpkgs-review-gha/actions/runs/19066454560


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>cryptpad</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>cryptpad</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = false)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>cryptpad</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = false)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>cryptpad</li>
  </ul>
</details>

eljamm commented:
There is a [hash mismatch on Darwin](https://github.com/eljamm/nixpkgs-review-gha/actions/runs/19066454560/job/54458230741#step:6:399), despite the sandbox being off.

themadbit commented on pkgs/by-name/cr/cryptpad/package.nix:
```diff
@@ -84,10 +118,10 @@ buildNpmPackage {
     repo = "cryptpad";
     rev = version;
-    hash = "sha256-NxkVMsfLzdzifdn+f0C6mBJGd1oLwcMTAIXv+gBG7rI=";
+    hash = "sha256-R8Oonrnb1tqvl1zTkWv5Xv/f8bFtUljD6X/re72IvsU=";
```
Thanks for the review @eljamm. @martinetd I have time, and I am working on the suggestion at the moment.
Yes, it would be better if we got it merged and iterate on the newer version and the rest of the nuances.

eljamm reviewed (approved):
Looks good!

martinetd reviewed (approved):
Didn't even cause a rebuild :)

Thanks for the update!

I'll open a follow-up PR for 2025.9 once this is merged
